### PR TITLE
Added Create, Update and Delete for User

### PR DESCRIPTION
**Pull request by christian1dk**

### CODE REVIEW

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -13,4 +15,86 @@ function userView() {
...
+            }
+            return redirect('/user/ret')->with('error', 'Brugeren eksisterer');          
+        }
+
+        $user = User::find($input['id']);
+        $user->first_name = $input['first_name'];
+        $user->last_name = $input['last_name'];
+        $user->role_id = $input['role_id'];
+        $user->email = $input['email'];
+        if (!empty($input['password'])) {
```
Comment (Yoropiata) [inline app/Http/Controllers/UserController.php]: I think this password change should be handled by a specific request to change password that requires entering the current password.

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -13,4 +15,86 @@ function userView() {
...
+        return redirect('/users');
+    }
+
+    /** 
+     * Delete User 
+     * 
+     * @return \Illuminate\Http\Response 
+     */ 
+    public function delete(Request $request) 
+    {
```
Comment (Yoropiata) [inline app/Http/Controllers/UserController.php:116]: we need a force delete endpoint too for admin

File: routes/web.php [hunk 3]
```diff
@@ -25,14 +25,18 @@
...
+    Route::get('/user/{id}/passwords', [UserGroupController::class, "myPasswordsView"])->whereNumber('id');
     
     Route::get('/', [PasswordController::class, "notificationPasswordsView"]);
     Route::get('/password', [PasswordController::class, "allPasswordsView"]);
     
+    //User
     Route::get('/users', [UserController::class, "userView"]);
-    Route::get('/user/{id}', [UserController::class, "profileView"]);
+    Route::get('/user/{id}', [UserController::class, "profileView"])->whereNumber('id');
+    Route::post('/user/create', [UserController::class, "create"]);
```
Comment (Yoropiata) [inline routes/web.php]: create is inferred by the post command. I recommend using `/user`

File: routes/web.php [hunk 4]
```diff
@@ -25,14 +25,18 @@
...
+    Route::put('/user/update', [UserController::class, "update"]);
```
Comment (Yoropiata) [inline routes/web.php]: update is inferred by put, however user id should be attached, I recommend `/user/{id}`

File: routes/web.php [hunk 5]
```diff
@@ -25,14 +25,18 @@
...
+    Route::delete('/user/delete',[UserController::class, "delete"]);
```
Comment (Yoropiata) [inline routes/web.php]: delete is inferred by delete request type, however user id should be attached, I recommend `/user/{id}`

For the force delete endpoint, I recommend `/user/{id}/force` request must be of type `delete`

**Review by Yoropiata — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/UserController.php (hunk 2 above)
Comment (christian1dk) [reply app/Http/Controllers/UserController.php:116]: We need a soft delete endpoint, but we need that option in the database

File: routes/web.php (hunk 4 above)
Comment (christian1dk) [reply routes/web.php:38]: Would we send that in the url?
We use request now.

File: routes/web.php (hunk 5 above)
Comment (christian1dk) [reply routes/web.php:39]: Would we send that in the url?
We use request now.

File: routes/web.php (hunk 4 above)
Comment (christian1dk) [reply routes/web.php]: what is the difference if we send it as /user/{id} or /user?{id}

File: routes/web.php (hunk 5 above)
Comment (christian1dk) [reply routes/web.php]: what is the difference if we send it as /user/{id} or /user?{id}

**Review by numtjok — APPROVED** [review]

**Review by Yoropiata — APPROVED** [review]

**Pull request merged**